netrd/distance/graph_diffusion_distance.py

```diff
@@ -0,0 +1,156 @@
+"""
+gdd.py
+--------------------------
+
+Graph diffusion distance, from
+
+Hammond, D. K., Gur, Y., & Johnson, C. R. (2013, December). Graph diffusion
+distance: A difference measure for weighted graphs based on the graph Laplacian
+exponential kernel. In Global Conference on Signal and Information Processing,
+2013 IEEE (pp 419-422). IEEE. https://doi.org/10.1109/GlobalSIP.2013.6736904
+
+author: Brennan Klein
+email: brennanjamesklein at gmail dot com
+Submitted as part of the 2019 NetSI Collabathon.
+
+"""
+
+import numpy as np
+import networkx as nx
+from scipy.sparse.csgraph import laplacian
+from .base import BaseDistance
+from ..utilities import undirected
+
+
+class GraphDiffusion(BaseDistance):
+    """Find the maximally dissimilar diffusion kernels between two graphs."""
+
+    @undirected
+    def dist(self, G1, G2, thresh=1e-14, resolution=1000):
+        r"""The graph diffusion distance between two graphs, :math:`G` and :math:`G'`,
+        is a distance measure based on the notion of flow within each graph. As
+        such, this measure uses the unnormalized Laplacian matrices of both
+        graphs, :math:`\mathcal{L}` and :math:`\mathcal{L}'`, and uses them to
+        construct time-varying Laplacian exponential diffusion kernels,
+        :math:`e^{-t\mathcal{L}}` and :math:`e^{-t\mathcal{L}'}`, by
+        effectively simulating a diffusion process for :math:`t` timesteps,
+        creating a column vector of node-level activity at each timestep. The
+        distance :math:`d_\texttt{GDD}(G, G')` is defined as the Frobenius norm
+        between the two diffusion kernels at the timestep :math:`t^{*}` where
+        the two kernels are maximally different.
```

Comment:
Question: confused about the definition of the distance. This paragraph could mean two things:

1. First run the diffusion for `t` steps. Then look for the timestep at which the Frobenius distance is largest. Return that. (This involves computing the Frobenius distance at each timestep and taking the maximum.)

2. First run the diffusion for `t` steps. Then return the Frobenius distance at the last timestep.

Reply:
Added a sentence clarifying that it's the former approach.